app/models/katello/repository.rb

```diff
@@ -641,6 +641,32 @@ def latest_dynflow_sync
                                 for_resource(self).order(:started_at).last
     end
 
+    def blocking_tasks
+      blocking_task_labels = [
+        ::Actions::Katello::Repository::Sync.name,
+        ::Actions::Katello::Repository::UploadFiles.name,
+        ::Actions::Katello::Repository::RemoveContent.name
+      ]
+      ForemanTasks::Task::DynflowTask.where(:label => blocking_task_labels)
+                                     .where.not(state: 'stopped')
+                                     .for_resource(self)
+                                     .order(:started_at)
+                                     .last
+    end
```

Comment:
And maybe one more thing, the method name is in plural, but it returns either a task or nil, that's slightly misleading